js/src/carousel.js

```diff
@@ -449,15 +444,18 @@ EventHandler.on(document, EVENT_CLICK_DATA_API, SELECTOR_DATA_SLIDE, function (e
 
   if (slideIndex) {
     carousel.to(slideIndex)
+    carousel._maybeEnableCycle()
     return
   }
 
   if (Manipulator.getDataAttribute(this, 'slide') === 'next') {
     carousel.next()
+    carousel._maybeEnableCycle()
     return
   }
 
   carousel.prev()
+  carousel._maybeEnableCycle()
```

Comment:
* `cycle` is called programmatically, and should always work triggering a  slide action.
* `_maybeEnableCycle` is called as a result of a user interaction, so it might not lead to a slide action.  
Inside `_slide`we do not know who called the method in order to initiate the cycle process. 

(saying all these, I wish I do not miss something)

As for the places it is placed the  `_maybeEnableCycle` , Ii tried to keep it as close a possible to the initial code. 
We may keep it for now and we can revisit later for more improvement